app/gui2/src/components/CodeEditor.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+import { ref, watchEffect } from 'vue'
+import {useWindowEvent} from "@/util/events";
+
+const emit = defineEmits<{
+  codeUpdate: [code: string]
+}>()
+
+const content = ref('main = \n    2 + 2')
+const shown = ref(false)
```

Comment:
I think this is addressed, as the code editor is self-contained now. Le me know if I missed something here.

Reply:
I don't like that `CodeEditor` emits `codeUpdate` event, while not having an input as well. You are still using a hardcoded code string in the component instead. Not sure if that is intentional and you simply wanted to bind the data to store as a separate task. So, everything below is mostly about that store binding, your choice if you want to do it in this PR or should it be separated.

---

I think that the `CodeEditor` should access the graph store directly. You can access it the same way as it is done in graph component:
```ts
let graphStore = useGraphStore()
```
All components receive exactly the same instance of the store, as it is intentionally a shared global reactive state. You can publish the `textContent` ref from inside the store to use it here (read only access).

The emitted `codeUpdate` event isn't necessary. But if we would have one, then it should only signal that a code have been changed, and possibly send the change event itself (i.e. insert/delete action and its range).

---

Totally optional, but If that's helpful for you, you can split that `CodeEditor` into two:
- `ModuleCodeEditor` - store-connected "container component" which handles top-level layout and binding of the `graphStore` to view,
-  `CodeInput` - presentational component that worries about displaying text as code and generates update events on user action.

Though if you do the split, then existing Yjs editor bindings (e.g. Monaco) will probably be harder to use, since you will need direct access to the `Y.Text` instance to bind it to the editor view.